src/dbc.rs

```diff
@@ -186,142 +403,120 @@ mod tests {
-
-            rr_builder = rr_builder.add_spent_proof_share(
-                input.spend_key(),
-                SpentProofShare {
-                    spent_sig,
-                    spentbook_pks,
-                    spentbook_sig_share,
-                },
-            );
+        for (i, (mlsag, _mlsag_material)) in reissue_tx
+            .mlsags
+            .iter()
+            .zip(material.inputs.iter())
+            .enumerate()
+        {
+            let spent_proof_share =
+                spentbook.log_spent(mlsag.key_image.into(), reissue_tx.clone())?;
+            rr_builder = rr_builder.add_spent_proof_share(i, spent_proof_share);
         }
 
         let rr = rr_builder.build()?;
-        let reissue_share = genesis_node.reissue(rr).unwrap();
-        assert_eq!(tx_blinded, reissue_share.dbc_transaction);
+        let reissue_share = mint_node.reissue(rr)?;
+        assert_eq!(reissue_tx.hash(), reissue_share.transaction.hash());
 
         let (mint_key_set, mint_sig_share) =
-            reissue_share.mint_node_signatures.values().next().unwrap();
+            &reissue_share.mint_node_signatures.values().next().unwrap();
+
         let mint_sig = mint_key_set
             .combine_signatures(vec![mint_sig_share.threshold_crypto()])
             .unwrap();
 
-        let fuzzed_parents = BTreeSet::from_iter(
-            reissue_share
-                .dbc_transaction
-                .inputs
-                .iter()
-                .copied()
-                .skip(n_drop_parents.coerce()) // drop some parents
-                .chain(
-                    // add some random parents
-                    (0..n_add_random_parents.coerce())
-                        .into_iter()
-                        .map(|_| rand::random::<SpendKey>()),
-                ),
-        );
+        let fuzzed_amt_secrets =
+            AmountSecrets::from_amount(amount + extra_output_amount.coerce::<Amount>(), &mut rng8);
+        let dbc_amount = fuzzed_amt_secrets.amount();
 
-        let fuzzed_content = DbcContent::new(
-            fuzzed_parents,
-            amount + extra_output_amount.coerce::<Amount>(),
-            crate::bls_dkg_id().public_key_set.public_key(),
-            AmountSecrets::random_blinding_factor(),
-        )?;
+        let fuzzed_content = DbcContent::from((
+            owner_once.owner_base.clone(),
+            owner_once.derivation_index,
+            fuzzed_amt_secrets.clone(),
+        ));
 
-        let mut fuzzed_transaction_sigs: BTreeMap<SpendKey, (PublicKey, Signature)> =
+        let mut fuzzed_transaction_sigs: BTreeMap<KeyImage, (PublicKey, Signature)> =
             BTreeMap::new();
 
-        // Add valid sigs
+        let mint_pk = mint_node.key_manager().public_key_set()?.public_key();
         fuzzed_transaction_sigs.extend(
             reissue_share
                 .mint_node_signatures
-                .keys()
+                .into_iter()
```

Comment:
Seems like we don't need the values, can we keep `.keys()`?

Reply:
the into_iter() avoided the need to clone() each in_owner inside the map().

I've changed it to use into_keys() instead.